lib/src/subscription.dart

```diff
@@ -0,0 +1,284 @@
+
+  @override
+  bool moveNext() => ++_index < _subscriptions.length;
+}
+
+/// {@category Sync}
+enum SubscriptionSetState {
+  /// This subscription set has not been persisted and has not been sent to the server.
+  /// This state is only valid for MutableSubscriptionSets
+  _uncommitted, // ignore: unused_field
+
+  /// The subscription set has been persisted locally but has not been acknowledged by the server yet.
+  pending,
+
+  /// The server is currently sending the initial state that represents this subscription set to the client.
+  _bootstrapping, // ignore: unused_field
+
+  /// This subscription set is the active subscription set that is currently being synchronized with the server.
+  complete,
+
+  /// An error occurred while processing this subscription set on the server. Check error_str() for details.
+  error,
+
+  /// The server responded to a later subscription set to this one and this one has been
+  /// trimmed from the local storage of subscription sets.
+  superseded,
+}
+
+/// A collection representing the set of active subscriptions for a [Realm] instance.
+///
+/// This is used in combination with [FlexibleSyncConfiguration] to
+/// declare the set of queries you want to synchronize with the server. You can access and
+/// read the subscription set freely, but mutating it must happen in an [update]
+/// block.
+///
+/// Any changes to the subscription set will be persisted locally and be available the next
+/// time the application starts up - i.e. it's not necessary to subscribe for the same query
+/// every time. Updating the subscription set can be done while offline, and only the latest
+/// update will be sent to the server whenever connectivity is restored.
+///
+/// It is strongly recommended that you batch updates as much as possible and request the
+/// dataset your application needs upfront. Updating the set of active subscriptions for a
+/// Realm is an expensive operation serverside, even if there's very little data that needs
+/// downloading.
+/// {@category Sync}
+abstract class SubscriptionSet with IterableMixin<Subscription> {
+  Realm _realm;
+  SubscriptionSetHandle _handle;
+
+  SubscriptionSet._(this._realm, this._handle);
+
+  /// Finds an existing [Subscription] in this set by its query
+  ///
+  /// The [query] is represented by the corresponding [RealmResults] object.
+  /// Finds a subscription by query.
+  ///
+  /// If the Subscription set does not contain a subscription with the provided query,
+  /// return null
+  Subscription? find<T extends RealmObject>(RealmResults<T> query) {
+    return realmCore.findSubscriptionByQuery(this, query).convert(Subscription._);
+  }
+
+  /// Finds an existing [Subscription] in this set by name.
+  ///
+  /// If the Subscription set does not contain a subscription with the provided name,
+  /// return null
+  Subscription? findByName(String name) {
+    return realmCore.findSubscriptionByName(this, name).convert(Subscription._);
```

Comment:
So you prefer **not** to abstract this very common pattern, and use:
```
    final result = realmCore.findSubscriptionByName(this, name);
    return result == null ? null : Subscription._(result);
```
instead?

I disagree. I think we should use abstractions to avoid repetitive code like this.

Anyway, I will do the change.